### PR TITLE
Add  metadata to BigQuery events

### PR DESCRIPTION
**Pull request by rafax**
Add `flagged` metadata to BigQuery events (so we can calculate statistics on flagged requests). 

## Test plan

- tested locally

### CODE REVIEW

File: cmd/cody-gateway/internal/httpapi/completions/upstream.go [hunk 1]
```diff
@@ -46,9 +46,10 @@ type upstreamHandlerMethods[ReqT UpstreamRequest] struct {
 	// validateRequest can be used to validate the HTTP request before it is sent upstream.
 	// Returning a non-nil error will stop further processing and return the given error
 	// code, or a 400.
+	// Last parameter is a boolean indicating whether the request was flagged during validation.
```
Comment (mrnugget) [inline cmd/cody-gateway/internal/httpapi/completions/upstream.go:49]:
```suggestion
	// Last return value is a boolean indicating whether the request was flagged during validation.
```

File: cmd/cody-gateway/internal/httpapi/completions/upstream.go [hunk 2]
```diff
@@ -46,9 +46,10 @@ type upstreamHandlerMethods[ReqT UpstreamRequest] struct {
...
 	//
 	// The provided logger already contains actor context.
-	validateRequest func(context.Context, log.Logger, codygateway.Feature, ReqT) (int, error)
+	validateRequest func(context.Context, log.Logger, codygateway.Feature, ReqT) (int, error, bool)
```
Comment (mrnugget) [inline cmd/cody-gateway/internal/httpapi/completions/upstream.go:52]: I'd say that Go convention is to always have `error` be the last return value. `(T, bool, error)` is [quite common](https://sourcegraph.sourcegraph.com/search?q=context:sourcegraph+content:%22%2C+bool%2C+error%29%22&patternType=standard&sm=0&groupBy=repo), so I'd go with that.

```suggestion
	validateRequest func(context.Context, log.Logger, codygateway.Feature, ReqT) (int, bool, error)
```

**Review by mrnugget — CHANGES_REQUESTED** [review]

File: cmd/cody-gateway/internal/httpapi/completions/upstream.go (hunk 1 above)
Comment (rafax) [reply cmd/cody-gateway/internal/httpapi/completions/upstream.go:49]: Thanks, not sure what I had in mind by calling it a parameter.
Applied during refactoring.

File: cmd/cody-gateway/internal/httpapi/completions/upstream.go (hunk 2 above)
Comment (rafax) [reply cmd/cody-gateway/internal/httpapi/completions/upstream.go]: Fair point, looks like I'm still not super used to >2 return values.

File: cmd/cody-gateway/internal/httpapi/completions/upstream.go [hunk 3]
```diff
@@ -46,9 +46,10 @@ type upstreamHandlerMethods[ReqT UpstreamRequest] struct {
 	// validateRequest can be used to validate the HTTP request before it is sent upstream.
 	// Returning a non-nil error will stop further processing and return the given error
 	// code, or a 400.
+	// Second return value is a boolean indicating whether the request was flagged during validation.
 	//
 	// The provided logger already contains actor context.
-	validateRequest func(context.Context, log.Logger, codygateway.Feature, ReqT) (int, error)
+	validateRequest func(context.Context, log.Logger, codygateway.Feature, ReqT) (int, bool, error)
```
Comment (eseliger) [inline cmd/cody-gateway/internal/httpapi/completions/upstream.go:52]: Maybe this makes it a bit clearer as well? 
```suggestion
	validateRequest func(context.Context, log.Logger, codygateway.Feature, ReqT) (httpStatus int, flagged bool, _ error)
```

**Review by eseliger — APPROVED** [review]

File: cmd/cody-gateway/internal/httpapi/completions/upstream.go (hunk 3 above)
Comment (rafax) [reply cmd/cody-gateway/internal/httpapi/completions/upstream.go]: Implemented.

**Review by mrnugget — APPROVED** [review]

**Pull request merged**